tests/budget_accounting_test.py

```diff
@@ -48,5 +52,102 @@ def test_compute_budgets(self):
+        accountant.request_budget(NoiseKind.LAPLACE)
+        accountant.request_budget(NoiseKind.GAUSSIAN)
+        self.assertEqual(2, len(accountant._mechanisms))
+
+    def test_weighted_mechanism_noise(self):
+        epsilon = 0.8966032834308992
+        delta = 1e-5
+        expected_pipeline_noise = 10.87860107421875
+        mechanism_weight = 2
+        expected_mechanism_noise = expected_pipeline_noise / mechanism_weight
+
+        accountant = PLDBudgetAccountant(epsilon, delta, 1e-2)
+        mechanism1 = accountant.request_budget(NoiseKind.LAPLACE)
+        accountant._mechanisms[0].weight = mechanism_weight
+        mechanism2 = accountant.request_budget(NoiseKind.GAUSSIAN)
+        accountant._mechanisms[1].weight = mechanism_weight
+        accountant.compute_budgets()
+        self.assertEqual(expected_pipeline_noise, accountant.minimum_noise_std)
+        self.assertEqual(expected_mechanism_noise, mechanism1.noise)
+        self.assertEqual(expected_mechanism_noise, mechanism2.noise)
+
+    def test_compute_adjust_noise_for_epsilon(self):
+        expected_noise = 9.8619384765625
+        epsilon = 1.2100629328988848
+        delta = 1e-5
+        accountant = PLDBudgetAccountant(epsilon, delta, 1e-2)
+        for _ in range(0, 4):
+            accountant.request_budget(NoiseKind.LAPLACE)
+            accountant.request_budget(NoiseKind.GAUSSIAN)
+
+        compare_pld = accountant._compose_distributions(expected_noise)
+        self.assertAlmostEqual(epsilon,
+                               compare_pld.get_epsilon_for_delta(delta), 3)
+        accountant.compute_budgets()
+        self.assertEqual(expected_noise, accountant.minimum_noise_std)
+
+    def test_compute_adjust_noise_for_epsilon_weighted(self):
+        expected_noise = 9.87554931640625
+        epsilon = 2.4857963874870936
+        delta = 1e-5
+        accountant = PLDBudgetAccountant(epsilon, delta, 1e-2)
+        for _ in range(0, 4):
+            accountant.request_budget(NoiseKind.LAPLACE)
+            accountant.request_budget(NoiseKind.GAUSSIAN)
+
+        for mechanism_internal in accountant._mechanisms:
+            mechanism_internal.weight = 2
+
+        compare_pld = accountant._compose_distributions(expected_noise)
+        self.assertAlmostEqual(epsilon,
+                               compare_pld.get_epsilon_for_delta(delta), 3)
+
+        accountant.compute_budgets()
+        self.assertEqual(expected_noise, accountant.minimum_noise_std)
+
+    def test_compute_adjust_noise_for_epsilon_sensitivity(self):
+        expected_noise = 9.8876953125
+        epsilon = 2.4838584185571655
+        delta = 1e-5
+        accountant = PLDBudgetAccountant(epsilon, delta, 1e-2)
+        for _ in range(0, 4):
+            accountant.request_budget(NoiseKind.LAPLACE)
+            accountant.request_budget(NoiseKind.GAUSSIAN)
+
+        for mechanism_internal in accountant._mechanisms:
+            mechanism_internal.sensitivity = 2
```

Comment:
Please test with different sensitivities and weights for different mechanisms as well